python/ray/serve/_private/router.py

```diff
@@ -56,6 +59,9 @@ async def resolve_async_tasks(self):
             replacement_table = dict(zip(tasks, resolved))
             self.args, self.kwargs = scanner.replace_nodes(replacement_table)
 
+        # Make the scanner GCable to avoid memory leak
+        delete_scanner_instance(id(scanner))
+
```

Comment:
just call `del scanner` should be enough because it invokes the same code path.

Reply:
The motivation here is to hide the logic for the scanner operation, given the `delete_scanner_instance` is thin, i can change it for now